adapters/envista_il.js

```diff
@@ -0,0 +1,1572 @@
+        return done(null, []);
+      }
+      const $ = cheerio.load(body);
+      const location = $('#lblStation').text();
+      // replicate the export form
+      let form = {};
+      form['__EVENTTARGET'] = 'lnkExport';
+      form['__EVENTARGUMENT'] = '';
+      form['__VIEWSTATE'] = $('#__VIEWSTATE').attr('value');
+      form['__VIEWSTATEGENERATOR'] = $('#__VIEWSTATEGENERATOR').attr('value');
+      form['__EVENTVALIDATION'] = $('#__EVENTVALIDATION').attr('value');
+      form['ddlExport'] = 'XML';
+      form['lblCurrentPage'] = '1';
+
+      let exportLink;
+      try {
+        exportLink = $('#form1').attr('action');
+        // temp measure
+        exportLink = exportLink.replace('./NewGrid', 'NewGrid');
+      } catch (err) {
+        return done(null, []);
+      }
+
+      exportLink = source.url + exportLink;
+      const tasks = [exportStationXML(source, exportLink, form, location, minAvgPeriod, regionName)];
+      parallel(tasks, (err, results) => {
+        return done(err, results);
+      });
+    });
+  };
+};
+
+// do export from query result
+const exportStationXML = function (source, link, form, location, minAvgPeriod, regionName) {
+  return function (done) {
+    request.post({
+      url: link,
+      form: form
+    }, (err, res, body) => {
+      if (err || res.statusCode !== 200) {
+        return done(null, []);
+      }
+      formatData(source, body, link, location, minAvgPeriod, regionName, (measurements) => {
+        return done(null, measurements);
+      });
+    });
+  };
+};
+
+const formatData = function (source, data, link, location, minAvgPeriod, regionName, cb) {
+  try {
+    const $ = cheerio.load(data, { xmlMode: true });
+    let base = {
+      city: regionName,
+      location: location,
+      averagingPeriod: {unit: 'hours', value: 0.25},
+      attribution: [{
+        name: 'Israel Ministry of Environmental Protection',
+        url: source.url
+      }]
+    };
+    base = Object.assign(base, coordinates[location]);
+
+    let rFullDate = /(\d{2}\/\d{2}\/\d{4} \d{2}:\d{2})/;
+    let measurements = [];
+    $('data').filter(function (i, el) {
+      return rFullDate.test($(this).attr('value'));
+    }).each(function (i, el) {
+      let dateM = rFullDate.exec($(this).attr('value'));
+      let dateProp = getDate(dateM[0], source.timezone);
+      let rParamUnit = /(\w*)\[ ([\w\d]*)\]/i;
+      $(this).find('name').filter(function (i, el) {
+        let match = rParamUnit.exec($(this).text());
+        if (!match) { return false; }
+        return acceptableParameters.indexOf(match[1].toLowerCase()) >= 0;
+      }).each(function (i, el) {
+        let paramUnitM = rParamUnit.exec($(this).text());
+        let m = Object.assign({}, base);
+        if (m !== null) {
+          m.parameter = paramUnitM[1].toLowerCase();
+          m.date = dateProp;
+          m.coordinates = matchCoordinates(location);
+          m.unit = paramUnitM[2];
+          m.value = Number($(this).next().text());
```

Comment:
I saw a bunch of 0's in the dryrun results. I just want to make sure those are actual 0's coming from the source and not what we're putting in when there is no measurement coming from the source?